.chloggen/mackjmr_expose-sourceprovider.yaml

```diff
@@ -0,0 +1,30 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: internal/datadog
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: |
+  - created new module `github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog`
+  - packages `github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/hostmetadata`, `github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/clientutil` and `github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/scrub` have been moved into `github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog/hostmetadata`, `github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog/clientutil` and `github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog/scrub`. 
+  - create new package `github.com/open-telemetry/opentelemetry-collector-contrib/pkg/datadog/hostmetadata` which exposes `GetSourceProvider` from `github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog/hostmetadata`
```

Comment:
The only user facing change from this list is the last one, the rest are internal changes

Reply:
good call, updated in: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37668/commits/938ef863a797fc30d4aaec3325c5b560e35a2dda.